### ISSUE TITLE
Support for Apple Silicon - bad CPU type in executable: flashlight

### ISSUE THREAD

**Issue by tormoseng** [bamlab/flashlight#199]
I'm running MacOS 14.2.1 on an Apple Mac M3. And after running `curl https://get.flashlight.dev | bash`, I try:
```
$ flashlight measure
```

Then, my response is
```
bad CPU type in executable: flashlight
```

Is this due to missing support of Apple Silicon?

Note: I'm able to clone your repo and run `test` and `report` as guided in the contribution.md, but it would be nice to do it with only `$ flaslight`

**Comment by Almouro** [comment]
Hi, thanks for raising this!

Turns out so far, we've been relying on people having Rosetta installed to run the Flashlight executable as an Intel executable

But we should really have a proper arm64 executable for Apple silicon as well, we'll look into this 👍

**Comment by gabimoncha** [comment]
I have an M1 Max and flashlight is working on my device

**Comment by tormoseng** [comment]
> I have an M1 Max and flashlight is working on my device

Without Rosetta?

**Comment by gabimoncha** [comment]
yes, installed through normal iTerm session, without Rosetta

**Comment by chanphiromsok** [comment]
it too slow when profiler on M1

**Comment by tormoseng** [comment]
@gabimoncha I just now saw that you wrote `iTerm session`, so I tried to get and run flashlight on iTerm (and not the Mac terminal), but it still fails with the same message. I'm on M3 if that is different from M1?

**Comment by tormoseng** [comment]
Last update from me. I'm now using `iTerm` with Rosetta to run the flashlight commands, while my regular mac terminal runs as before. It works - even though supporting Apple Silicon would be nice!

**Comment by Randall71** [comment]
@Almouro This problem deserves to be pinned

**Comment by Almouro** [comment]
Good point @Randall71! Still not fixed on our side, it's just a bit annoying since we have to codesign the executable now, but it's underway

In the meantime I've pinned the issue and linked @Sumedh-k's helpful comment to the install docs in https://docs.flashlight.dev/
